### DISCUSSION TITLE
Extract ID (gallery number)  from file title as source

### DISCUSSION THREAD

**Discussion by Texugooftomorrow** [Difegue/LANraragi#938] [Q&A] [answered]
Suppose a file from gallery-dl is named 123456 - Ubuntu ISO | Canonical.cbz, by using the file name parsing plugin to extract the data, it messes up and  it incorrectly titles the work as 123456, uses the "Ubuntu ISO" as author and adds the "canonical" as the language or something, is there a way to parse the first numbers of the file name as the Source tag (from the site that uses numbers (nH)) so it can run the Source converter plugin to add the base url so then the files can use the nh plugin to batch look for the correct metadata? I didn't find any position in the file name that the ID can be put so it is properly recognized as the source tag, how should it be done?  I didn't find any documentation on the plugins itself.

**Comment by Difegue** [comment, marked as answer]
As mentioned in the nH plugin description:  
> Supports reading the ID from files formatted as \"{Id} Title\"  

I suppose you could mass-rename your gallery-dl downloads so that they fit this naming scheme -- In which case you won't need to use the filename parsing plugin.

**Reply by Texugooftomorrow** [reply]
So should I test renaming it as {123456} Ubuntu ISO | Canonical.cbz for example?

**Reply by Difegue** [reply]
Yep! that should work.

**Reply by Texugooftomorrow** [reply]
thanks!

**Reply by Texugooftomorrow** [reply]
Just tested, batch operations using the nH plugin returns:

Started Batch Operation...
************
Error while processing ID xxxxxxxxxxxxxx (No matching nH Gallery Found!)

************
(code 1000)

and do nothing, name scheme is {nhid} Name.cbz

**Reply by Difegue** [reply]
The plugin technically uses the title of the imported file instead of the filename at the moment - could you check that the name scheme carried over to the title properly?  

There's no need to use batch for this btw you can directly run plugins from the edit metadata page.

**Reply by Texugooftomorrow** [reply]
Yes, that was the problem, after I manually renamed the title in library it found the match no problem, is there a way to fix this naming for already imported files?

**Reply by Difegue** [reply]
A solution would be to use `$lrr_info->{file_path}` instead of `$lrr_info->{archive_title}` in the plugin...

But you'd probably have to tweak the regex a bit so that it discards all the folder info in the path, or add a call to `basename` before like what RegexParse does: https://github.com/Difegue/LANraragi/blob/26ff17a1f7c801723378f6f8cc2d75b8a9a8bce1/lib/LANraragi/Plugin/Metadata/RegexParse.pm#L50  
I might just implement that, it feels more useful to draw from the filename than the registered archive title.

**Reply by Texugooftomorrow** [reply]
Hi! In the end, I took advantage of it being in docker, and just rebuild the container with a new database, after running a bash script to batch rename the files in the correct syntax {ID} name.cbz (by detecting the first digits before a space and putting the contents inside a {}), then after initial scan and plugin run it detected everything that it could without problems. I think like you said that parsing from the file name would be a great addition instead from the database. As it's way easier to manipulate the file itself in cases like those.
